test/thorest-api/logs/query-events.test.ts

```diff
@@ -1,39 +1,627 @@
-import assert from 'node:assert'
-import { generateWalletWithFunds } from '../../../src/wallet'
+import {
+    getTransferDetails,
+    readRandomTransfer,
+    Transfer,
+} from '../../../src/populated-data'
+import { HEX_REGEX, HEX_REGEX_64 } from '../../../src/utils/hex-utils'
+import { components } from '../../../src/open-api-types'
+import { EventsContract__factory } from '../../../typechain-types'
+import { addAddressPadding } from '../../../src/utils/padding-utils'
+import { generateAddresses, ThorWallet } from '../../../src/wallet'
 
-describe('POST /logs/event', () => {
-    it('should find an event log', async () => {
-        // triggers a VTHO transfer event
-        const { receipt } = await generateWalletWithFunds()
-
-        const eventLogs = await Node1Client.queryEventLogs({
-            range: {
-                to: receipt.meta.blockNumber,
-                from: receipt.meta.blockNumber,
-                unit: 'block',
+const buildRequestFromTransfer = (
+    transfer: Transfer,
+): components['schemas']['EventLogFilterRequest'] => {
+    return {
+        range: {
+            to: transfer.meta.blockNumber,
+            from: transfer.meta.blockNumber,
+            unit: 'block',
+        },
+        options: {
+            offset: 0,
+            limit: 10_000,
+        },
+        criteriaSet: [
+            {
+                address: contractAddresses.energy,
             },
-            options: {
-                offset: 0,
-                limit: 100,
+        ],
+    }
+}
+
+describe('POST /logs/event', () => {
+    let transferDetails = getTransferDetails()
+
+    let wallet: ThorWallet = ThorWallet.new(true)
+
+    it('event log should be included in query', async () => {
+        const transfer = readRandomTransfer()
+
+        const request = buildRequestFromTransfer(transfer)
+
+        const eventLogs = await Node1Client.queryEventLogs(request)
+
+        expect(eventLogs.success, 'API response should be a success').toBeTrue()
+        expect(eventLogs.httpCode, 'Expected HTTP Code').toEqual(200)
+
+        const relevantLog = eventLogs.body?.find((log) => {
+            return log?.meta?.txID === transfer.meta.txID
+        })
+
+        expect(relevantLog, 'Expected event log response format').toEqual({
+            address: contractAddresses.energy,
+            topics: transfer.vtho.topics,
+            data: expect.stringMatching(HEX_REGEX_64),
+            meta: {
+                blockID: expect.stringMatching(HEX_REGEX_64),
+                blockNumber: expect.any(Number),
+                blockTimestamp: expect.any(Number),
+                txID: expect.stringMatching(HEX_REGEX_64),
+                txOrigin: transfer.meta.txOrigin,
+                clauseIndex: expect.any(Number),
             },
-            criteriaSet: [
-                {
-                    address: contractAddresses.energy,
+        })
+    })
+
+    describe('query by "range"', () => {
```

Comment:
should we try with range set to null - seems like a valid scenario from API perspective